### PR TITLE
Chatbot route added

### LINKED ISSUE #9: Add descriptive route names
Switch from using placeholder names to routes that can be used in production

### LINKED ISSUE #21: create route that calls chatbot function

### PR DESCRIPTION
**Pull request by svhnstorage**
Cleaned up route names, added a route that calls the chatbot.

Issue: Closes #9

### Frontend:
- 
### Backend:
-Added chatbot route
-Cleaned up names

### CODE REVIEW

File: backend/routes.py [hunk 1]
```diff
@@ -1,30 +1,42 @@
...
             
-#endpoint POST
-@app.route('/sentences', methods = ['GET','POST'])
-def test_route():
-    sentence = {'sass' : request.json['sass']}
-
-    #sentences.append(sentence)
-    return jsonify({'sentences' : sentence})
+#endpoint that echos input (Used For Testing)
+@app.route('/repeat', methods = ['GET','POST'])
```
Comment (JasperCheung) [inline backend/routes.py:24]: Just make this a `POST`?

File: backend/routes.py [hunk 2]
```diff
@@ -1,30 +1,42 @@
...
+def repeat_route():
+        if request.method == 'POST':
```
Comment (JasperCheung) [inline backend/routes.py:26]: Just make this a post so you don't need the if statement?

File: backend/routes.py [hunk 3]
```diff
@@ -1,30 +1,42 @@
...
+                user_inp = request.get_json()
+                return jsonify(user_inp)
+        return jsonify({"get" : "requested"})
+
+#endpoint that calls chatbot function (SHRDLU Response)
+@app.route('/chat', methods = ['GET', 'POST'])
+def chatbot_route():
+        if request.method == 'POST':
```
Comment (JasperCheung) [inline backend/routes.py:35]: Just make this a post so you don't need the if statement?

File: backend/routes.py [hunk 4]
```diff
@@ -1,30 +1,42 @@
...
-    sentence = {'sass' : request.json['sass']}
-
-    #sentences.append(sentence)
-    return jsonify({'sentences' : sentence})
+#endpoint that echos input (Used For Testing)
+@app.route('/repeat', methods = ['GET','POST'])
+def repeat_route():
+        if request.method == 'POST':
+                user_inp = request.get_json()
+                return jsonify(user_inp)
```
Comment (JasperCheung) [inline backend/routes.py]: Can you make this match chatbot route?              
`return jsonify({"SHRDLU: ": user_inp)`

File: backend/routes.py [hunk 5]
```diff
@@ -1,30 +1,42 @@
...
-def test_route():
-    sentence = {'sass' : request.json['sass']}
-
-    #sentences.append(sentence)
-    return jsonify({'sentences' : sentence})
+#endpoint that echos input (Used For Testing)
+@app.route('/repeat', methods = ['GET','POST'])
+def repeat_route():
+        if request.method == 'POST':
+                user_inp = request.get_json()
```
Comment (JasperCheung) [inline backend/routes.py]: Can you make this match chatbot route? So require a `"user"` field.

```
post_data = request.get_json()
user_res = post_data["user"]
```

**Review by JasperCheung — CHANGES_REQUESTED** [review]
Looks good, added some changes. Can you update the documentation to match these new routes? 

@joshcwinton you can match the calls here from on that frontend ticket.

File: backend/routes.py (hunk 2 above)
Comment (svhnstorage) [reply backend/routes.py:26]: might need get for testing to see if the endpoint is fine

File: backend/routes.py (hunk 3 above)
Comment (svhnstorage) [reply backend/routes.py:35]: might need get for testing to see if the endpoint is fine

File: backend/routes.py (hunk 4 above)
Comment (svhnstorage) [reply backend/routes.py]: done

File: backend/routes.py (hunk 5 above)
Comment (svhnstorage) [reply backend/routes.py]: done

File: backend/routes.py [hunk 6]
```diff
@@ -1,30 +1,43 @@
...
+@app.route('/repeat', methods = ['GET','POST'])
+def repeat_route():
+        if request.method == 'POST':
+                post_data = request.get_json()
+                user_res = post_data["user"]
+                return jsonify({"SHRDLU Echo: " : user_res})
+        return jsonify({"get" : "requested"})
+
+#endpoint that calls chatbot function (SHRDLU Response)
+@app.route('/chat', methods = ['GET', 'POST'])
```
Comment (JasperCheung) [inline backend/routes.py:33]: I'll just make a note here to test in the future.

**Review by JasperCheung — APPROVED** [review]
LGTM.

File: backend/routes.py [hunk 7]
```diff
@@ -1,30 +1,43 @@
...
-
-    #sentences.append(sentence)
-    return jsonify({'sentences' : sentence})
+#endpoint that echos input (Used For Testing)
+@app.route('/repeat', methods = ['GET','POST'])
+def repeat_route():
+        if request.method == 'POST':
+                post_data = request.get_json()
+                user_res = post_data["user"]
+                return jsonify({"SHRDLU Echo: " : user_res})
```
Comment (JasperCheung) [inline backend/routes.py:29]: I like the Echo, nice touch.

**Pull request merged**